qemu/tests/watchdog.py

```diff
@@ -87,9 +87,17 @@ def _action_check(test, session, watchdog_action):
         response_timeout = int(params.get("response_timeout", '240'))
         error_context.context("Check whether or not watchdog action '%s' took"
                               " effect" % watchdog_action, logging.info)
+        if watchdog_action == "inject-nmi":
+            if not utils_misc.wait_for(lambda: "NMI received" in session.get_output(),
+                                       response_timeout, 0, 1):
+                test.fail("Guest didn't receive debug info with 'NMI received',"
+                          "so action '%s' didn't take effect." % watchdog_action)
+            msg = session.get_output().splitlines()[-8:]
+            logging.info("Guest received debug info: %s" % msg)
         if not utils_misc.wait_for(lambda: not session.is_responsive(),
                                    response_timeout, 0, 1):
-            if watchdog_action == "none" or watchdog_action == "debug":
+            if (watchdog_action == "none" or watchdog_action == "debug" or
+                    watchdog_action == "inject-nmi"):
```

Comment:
"inject-nmi" has checked 'session.get_output()' within 'response_timeout', it seems not make sense to have another 'is_responsive()' check in one more 'response_timeout'.

Reply:
Line 91 means session is responsive until NMI is recieved, and Line 99 can test if session is responsive after watchodog fires.

Reply:
`if (watchdog_action in ("none",  "debug", injecr-nim))` maybe more clear

Reply:
Suggest to update line85 too if you want to add another scenario, but it is ok for me.

Reply:
Sorry, ignore me.